### PR TITLE
Email Reporting

### PR DESCRIPTION
**Pull request by vijay1275**
This PR includes the following things:
    * create html reports
    * parse html reports
    * generate run_id for every test run
    * sending emails about test run results

### CODE REVIEW

File: ocs/ocp.py [hunk 1]
```diff
@@ -259,6 +259,34 @@ def wait_for_resource(
...
+            selector (str): The label selector to look for
+            all_namespaces (bool): Equal to oc get <resource> -A
+
+        Example:
+            describe(pod, csi-cephfsplugin-provisioner-0)
+
+        Returns:
+            dict: Dictionary represents a returned yaml file
+        """
+        command = f"get {self.kind} {resource_name}"
```
Comment (RazTamir) [inline ocs/ocp.py]: get should be replaced with describe?

File: ocs/ocp.py [hunk 2]
```diff
@@ -259,6 +259,34 @@ def wait_for_resource(
 
         return False
 
+    def describe(
+        self, resource_name='', out_yaml_format=True, selector=None,
```
Comment (RazTamir) [inline ocs/ocp.py]: can you describe a resource without providing its name?

File: ocsci/pytest_customization/ocscilib.py [hunk 3]
```diff
@@ -70,6 +82,32 @@ def pytest_configure(config):
     """
     if not config.getoption("--help"):
         process_cluster_cli_params(config)
+        # Add OCS related versions to the html report and remove extraneous metadata
+        print("Collecting Cluster versions .....")
```
Comment (RazTamir) [inline ocsci/pytest_customization/ocscilib.py]: remove '.....'

**Comment by vavuthu** [comment]
Closing this PR as it opened from another account

File: ocsci/pytest_customization/ocscilib.py [hunk 4]
```diff
@@ -126,6 +164,10 @@ def process_cluster_cli_params(config):
     ocsci_config.RUN['cli_params']['deploy'] = get_cli_param(config, "deploy", default=False)
     ocsci_config.ENV_DATA['cluster_name'] = cluster_name
     ocsci_config.ENV_DATA['cluster_path'] = cluster_path
+    ocsci_config.ENV_DATA['email'] = get_cli_param(config, 'email')
+    ocsci_config.ENV_DATA['report'] = get_cli_param(config, '--html')
+    if ocsci_config.ENV_DATA['email'] and not ocsci_config.ENV_DATA['report']:
+        pytest.exit("--html option must be provided to send email reports")
```
Comment (RazTamir) [inline ocsci/pytest_customization/ocscilib.py]: Why not add the --html here?

File: utility/utils.py [hunk 5]
```diff
@@ -735,3 +740,187 @@ def is_cluster_running(cluster_path):
...
+    Decomposes the given html attributes
+
+    Args:
+        soup (obj): BeautifulSoup object
+        attributes (list): attributes to decompose
+
+    Returns: None
+
+    """
+
```
Comment (RazTamir) [inline utility/utils.py]: remove empty line

File: utility/utils.py [hunk 6]
```diff
@@ -735,3 +740,187 @@ def is_cluster_running(cluster_path):
...
+    for email reporting
+
+    Args:
+        soup (obj): BeautifulSoup object
+
+    """
+    decompose_html_attributes(soup, ["extra", "col-links"])
+    soup.find(id="not-found-message").decompose()
+
+    for tr in soup.find_all('tr'):
```
Comment (RazTamir) [inline utility/utils.py]: please use meaningful names 'tr' - fix all below

**Pull request closed without merging**